### PR TITLE
Instantiate array/hashmaps holders right away for listener use

### LINKED ISSUE #5408: NPE adding picture from WP media library
- Go to WP Media Library
- Select two or more pictures (Among of those selected pictures should be at least one in the "retry" state).
- Tap the pencil icon --> Crash.

```
03-15 10:35:27.702 20283-20283/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: org.wordpress.android, PID: 20283
                                                                       java.lang.NullPointerException
                                                                           at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:990)
                                                                           at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:985)
                                                                           at org.wordpress.android.editor.EditorFragment.appendMediaFile(EditorFragment.java:1022)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.addExistingMediaToEditor(EditPostActivity.java:1302)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.prepareMediaPost(EditPostActivity.java:1491)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.fillContentEditorFields(EditPostActivity.java:1422)
                                                                           at org.wordpress.android.ui.posts.EditPostActivity.onEditorFragmentInitialized(EditPostActivity.java:2241)
                                                                           at org.wordpress.android.editor.EditorFragment.onCreateView(EditorFragment.java:334)
                                                                           at android.app.Fragment.performCreateView(Fragment.java:2353)
                                                                           at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:995)
                                                                           at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1171)
                                                                           at android.app.BackStackRecord.run(BackStackRecord.java:816)
                                                                           at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1578)
                                                                           at android.app.FragmentManagerImpl.executePendingTransactions(FragmentManager.java:563)
                                                                           at android.support.v13.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:153)
                                                                           at android.support.v4.view.ViewPager.populate(ViewPager.java:1268)
                                                                           at android.support.v4.view.ViewPager.populate(ViewPager.java:1116)
                                                                           at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1642)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
                                                                           at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6083)
                                                                           at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
                                                                           at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:139)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6083)
                                                                           at android.support.v7.widget.ActionBarOverlayLayout.onMeasure(ActionBarOverlayLayout.java:393)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6083)
                                                                           at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6083)
                                                                           at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1464)
                                                                           at android.widget.LinearLayout.measureVertical(LinearLayout.java:758)
                                                                           at android.widget.LinearLayout.onMeasure(LinearLayout.java:640)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6083)
                                                                           at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
                                                                           at com.android.internal.policy.DecorView.onMeasure(DecorView.java:689)
                                                                           at android.view.View.measure(View.java:19857)
                                                                           at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2275)
                                                                           at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1366)
                                                                           at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1619)
                                                                           at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
                                                                           at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6337)
                                                                           at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
                                                                           at android.view.Choreographer.doCallbacks(Choreographer.java:686)
                                                                           at android.view.Choreographer.doFrame(Choreographer.java:621)
                                                                           at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
                                                                           at android.os.Handler.handleCallback(Handler.java:751)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                           at android.os.Looper.loop(Looper.java:154)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
03-15 10:35:27.702 20283-20283/org.wordpress.android E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                           at com.android.internal.os.ZygoteIn
```

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #5408 

To test:
- Go to WP Media Library
- Select two or more pictures (Among of those selected pictures should be at least one in the "retry" state).
- Tap the pencil icon --> Crash.

cc @daniloercoli

### CODE REVIEW

**Comment by daniloercoli** [comment]
Oh cool! I had just assigned myself to the issue.
What happens to the picture in the "retry" state?

**Comment by daniloercoli** [comment]
Unfortunately the issue is still there, since the URL of the media in the "retry" state, or "queued" state, is not yet available. It's still not on the server.
```
03-15 19:56:41.950 32694-32694/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: org.wordpress.android, PID: 32694
                                                   java.lang.NullPointerException
                                                       at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:990)
                                                       at java.util.concurrent.ConcurrentHashMap.put(ConcurrentHashMap.java:985)
                                                       at org.wordpress.android.editor.EditorFragment.appendMediaFile(EditorFragment.java:1018)
                                                       at org.wordpress.android.ui.posts.EditPostActivity.addExistingMediaToEditor(EditPostActivity.java:1301)
                                                       at org.wordpress.android.ui.posts.EditPostActivity.prepareMediaPost(EditPostActivity.java:1490)
                                                       at org.wordpress.android.ui.posts.EditPostActivity.fillContentEditorFields(EditPostActivity.java:1421)
                                                       at org.wordpress.android.ui.posts.EditPostActivity.onEditorFragmentInitialized(EditPostActivity.java:2243)
                                                       at org.wordpress.android.editor.EditorFragment.onCreateView(EditorFragment.java:330)
                                                       at android.app.Fragment.performCreateView(Fragment.java:2353)
```

I guess we should, for a quick fix in `7.0`, just check the state of each picture and show a prompt to the user if the picture is not yet uploaded to the server. Maybe in a future version we can even upload the missing pictures. @nbradbury Thoughts on this?

**Comment by mzorz** [comment]
It should be fixed now! check commit b60d352 @daniloercoli

**Comment by nbradbury** [comment]
> I guess we should, for a quick fix in 7.0, just check the state of each pictures and show a prompt to the user if the picture is not yet uploaded to the server. Maybe in a future version we can even upload the missing pictures. @nbradbury Thoughts on this?

In the short term we could disable the pencil icon when a failed image is selected, since it doesn't make sense to create a post using an image that failed to upload. 

In the long term, I'd vote to get rid of the pencil icon altogether and make the media browser dedicated to managing your media. It seems weird to me that users would go to the media browser to create a new post in the first place.

**Comment by daniloercoli** [comment]
> In the short term we could disable the pencil icon when a failed image is selected, since it doesn't make sense to create a post using an image that failed to upload.

Makes sense to me.
@mzorz What do you think about disabling the pencil icon? If that's OK for you we can go this way.

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java [hunk 1]
```diff
@@ -115,10 +115,11 @@
     private boolean mHideActionBarOnSoftKeyboardUp = false;
     private boolean mIsFormatBarDisabled = false;
 
-    private ConcurrentHashMap<String, MediaFile> mWaitingMediaFiles;
-    private Set<MediaGallery> mWaitingGalleries;
-    private Map<String, MediaType> mUploadingMedia;
-    private Set<String> mFailedMediaIds;
+    private ConcurrentHashMap<String, MediaFile> mWaitingMediaFiles = new ConcurrentHashMap<>();
```
Comment (daniloercoli) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java]: Is it necessary to instantiate these variables here? The previous solution was working right, no?

**Comment by mzorz** [comment]
> @mzorz What do you think about disabling the pencil icon? If that's OK for you we can go this way.
sure thing, I agreed last night with my 👍 emoji 😁 
Will try in another commit

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java (hunk 1 above)
Comment (mzorz) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java]: yes, but it doesn't do any harm and it reassures the variables will be created even if the listener is referenced elsewhere, so I think it's safer to keep the initializers here

**Comment by mzorz** [comment]
> In the long term, I'd vote to get rid of the pencil icon altogether and make the media browser dedicated to managing your media. It seems weird to me that users would go to the media browser to create a new post in the first place.

addressed this here 17a0514

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**